core/commands/add.go

```diff
@@ -107,10 +107,20 @@ You can now check what blocks have been created by:
   QmerURi9k4XzKCaaPbsK6BL5pMEjF7PGphjDvkkjDtsVf3 868
   QmQB28iwSriSUSMqG2nXDTLtdPHgWb4rebBrU7Q1j4vxPv 338
 
-Finally, a note on hash determinism. While not guaranteed, adding the same
-file/directory with the same flags will almost always result in the same output
-hash. However, almost all of the flags provided by this command (other than pin,
-only-hash, and progress/status related flags) will change the final hash.
+Finally, a note on hash (CID) determinism and 'ipfs add' command.
+
+Given the same input and same add parameters, you will ALWAYS get the same CID.
+What is not guaranteed is for the implicit defaults of 'ipfs add' to remain the
+same in future Kubo releases, or for other IPFS software to use the same import
+parameters as Kubo.
```

Comment:
Given that this is not the first time we've clarified this it might be better to be even more explicit that `ipfs add` across versions resulting in the same CID is not supported behavior.

For example, say there's a new flag we find ourselves adding in the future that isn't represented now.

As an example, say we added support for UnixFS v1.5 and added support for metadata via `ipfs add` by default (IMO a bad idea, but beside the point here). There is currently no flag for `--disable-metadata` because that feature doesn't exist. Therefore, even though I diligently wrote down "To Self: Always use `ipfs add --chunker=buzhash --cid-version=1 --raw-leaves --hash=sha3-256 --inline=true --inline-limit=37 --trickle -r`" it's still insufficient because I didn't realize to add `--disable-metadata=true` because that flag didn't exist yet. 

I'd be clear here and say something like `ipfs add` will generate the same CID for the same data on the same version of Kubo "only", and that while the parameters of this function do not tend to change frequently or significantly the Kubo developers reserve the right to change various defaults so as to give users a better experience. If you insist on being able to record the exact parameters you should also include the version of Kubo used to do the import. Then nudge people towards CAR files (or pinning DAGs on other nodes or backing up the repo) for backups.

Reply:
But a weaker promise would be very valuable for operators of IPFS services: Can CIDs generated with an older Kubo version always be reproduced with a newer version, with possibly different flags?

Rationale: While it may be infeasible to install a 10-years-old Kubo to recreate CIDs from file system directories, setting the proper flags of a _current_ Kubo is no problem.

Reply:
@bronger you could use a 10 year old `go-car` instead. (we would need to add more options to the chunker but it's details).
Just pipe the cars into `ipfs dag import`.

I like this option much better as it's far easier to maintain, and we remove all consistency promises from kubo.

Reply:
I've updated this PR with a shorter note (without going into all the intricacies of import parameter space)
See https://github.com/ipfs/kubo/pull/9128/commits/339285fb133dd8222cde0755b7eebe7b44559671 

@aschmahmann  Do you feel it is sufficient?